wolfcrypt/src/port/arm/armv8-sha256.c

```diff
@@ -110,159 +110,204 @@ int wc_Sha256Update(Sha256* sha256, const byte* data, word32 len)
 {
 
     /* do block size increments */
-    byte* local = (byte*)sha256->buffer;
+    word32 add = min(len, SHA256_BLOCK_SIZE - sha256->buffLen);
+    word32 numBlocks;
 
-    while (len) {
-        word32 add = min(len, SHA256_BLOCK_SIZE - sha256->buffLen);
-        XMEMCPY(&local[sha256->buffLen], data, add);
+    /* fill leftover buffer with data */
+    XMEMCPY((byte*)(sha256->buffer) + sha256->buffLen, data, add);
+    sha256->buffLen += add;
+    data            += add;
+    len             -= add;
 
-        sha256->buffLen += add;
-        data            += add;
-        len             -= add;
+    /* number of blocks in a row to complete */
+    numBlocks = (len + sha256->buffLen)/SHA256_BLOCK_SIZE;
 
-        if (sha256->buffLen == SHA256_BLOCK_SIZE) {
-            word32* Kpt = (word32*)K;
-            word32* bufferPt = sha256->buffer;
-            word32* digestPt = sha256->digest;
+    if (numBlocks > 0) {
+        word32* bufferPt = sha256->buffer;
+        word32* digestPt = sha256->digest;
```

Comment:
With the constraint I was using "r" (register operand allowed if it is in a general register) was needed. Changed the constraint to "m" (A memory operand is allowed, with any kind of address that the machine supports in general) and am then able to remove these pointers and use just memory addresses.